File: spec/javascript/test_instance_std_array.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('InstanceStdArray', 'src/instance_std_array.bin', function(r) {
    assert.equal(r.ofs, 0x10);
    assert.equal(r.qtyEntries, 3);
    assert.equal(r.entrySize, 4);

    assert.equal(r.entries.length, 3);
    assert.equal(r.entries[0].toString(), new Uint8Array([0x11, 0x11, 0x11, 0x11]).toString());
    assert.equal(r.entries[1].toString(), new Uint8Array([0x22, 0x22, 0x22, 0x22]).toString());
    assert.equal(r.entries[2].toString(), new Uint8Array([0x33, 0x33, 0x33, 0x33]).toString());
});

File: spec/java/src/io/kaitai/struct/spec/TestInstanceStdArray.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.*;
import org.testng.annotations.Test;

import java.nio.charset.Charset;
import java.util.ArrayList;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertEqualsNoOrder;

public class TestInstanceStdArray extends CommonSpec {
    @Test
    public void testInstanceStdArray() throws Exception {
        InstanceStdArray r = InstanceStdArray.fromFile(SRC_DIR + "instance_std_array.bin");

        assertEquals(r.ofs(), 0x10);
        assertEquals(r.qtyEntries(), 3);
        assertEquals(r.entrySize(), 4);

        assertEquals(r.entries().size(), 3);
        assertEquals(r.entries().get(0), new byte[] { 0x11, 0x11, 0x11, 0x11 });
        assertEquals(r.entries().get(1), new byte[] { 0x22, 0x22, 0x22, 0x22 });
        assertEquals(r.entries().get(2), new byte[] { 0x33, 0x33, 0x33, 0x33 });
    }

}
